File: patch-version.py
import json, os, re

semver = os.environ['SEMVER']
pubkey = os.environ.get('PUBKEY', '')

# tauri.conf.json
with open('src-tauri/tauri.conf.json') as f:
    c = json.load(f)
c['version'] = semver
if pubkey and c.get('plugins', {}).get('updater'):
    c['plugins']['updater']['pubkey'] = pubkey
elif not pubkey and c.get('plugins', {}).get('updater'):
    del c['plugins']['updater']
with open('src-tauri/tauri.conf.json', 'w') as f:
    json.dump(c, f, indent=2)
print('tauri.conf.json version =', semver)

# Cargo.toml
with open('src-tauri/Cargo.toml') as f:
    t = f.read()
t = re.sub(r'^version = "[^"]+"', 'version = "' + semver + '"', t, flags=re.MULTILINE)
with open('src-tauri/Cargo.toml', 'w') as f:
    f.write(t)
print('Cargo.toml version =', semver)

File: patch-version.js
const fs     = require('fs');
const semver = process.env.SEMVER;
const pubkey = process.env.PUBKEY || '';

if (!semver) { console.error('SEMVER not set'); process.exit(1); }

// tauri.conf.json
const confPath = 'src-tauri/tauri.conf.json';
const c = JSON.parse(fs.readFileSync(confPath, 'utf8'));
c.version = semver;
if (pubkey && c.plugins && c.plugins.updater) {
  c.plugins.updater.pubkey = pubkey;
} else if (!pubkey && c.plugins && c.plugins.updater) {
  delete c.plugins.updater;
}
fs.writeFileSync(confPath, JSON.stringify(c, null, 2));
console.log('tauri.conf.json version =', semver);

// Cargo.toml
const cargoPath = 'src-tauri/Cargo.toml';
let cargo = fs.readFileSync(cargoPath, 'utf8');
cargo = cargo.replace(/^version = "[^"]+"/m, 'version = "' + semver + '"');
fs.writeFileSync(cargoPath, cargo);
console.log('Cargo.toml version =', semver);
